go/libraries/doltcore/table/typed/json/reader.go

```diff
@@ -77,8 +77,13 @@ func NewJSONReader(nbf *types.NomsBinFormat, r io.ReadCloser, info *JSONFileInfo
 		return nil, err
 	}
 
+	ableToDecode := true
```

Comment:
It was tricky to verify the schema for JSON because either a schema file is required or the schema from the existing table is used. When decoding the JSON rows here, an error is returned if the rows don't match the existing schema. This was not being handled before so I added an `AbleToDecode` property to `JSONFileInfo` that tracks whether the rows were decoded or not and can be used in the `VerifySchema` method. Let me know if you think there's a better way to do this - I couldn't directly compare the schemas here like I could for CSV and XLSX files.